notebooks/gridded_stitching_regular-py.py

```diff
@@ -0,0 +1,208 @@
+    for f in fl:
+        o = load_data(f)
+        data_list.append(o)
+
+    return data_list
+
+
+def stitch_gridded(rp, dl, fl):
+    """stitch the gridded data together.
+
+        :param rp:             data frame of the recepies
+        :param dl:             list of the data files
+        :param fl:             list of the data file names
+        :return:               xarray data set
+
+        TODO add a check to make sure that there is only one stitching id being passed into
+        the function.
+    """
+
+    rp.reset_index(drop=True, inplace=True)
+    out = get_netcdf_values(i=0, dl=dl, rp=rp, fl=fl)
+
+    for i in range(1,len(rp)):
+        new_vals = get_netcdf_values(i=i, dl=dl, rp=rp, fl=fl)
+        out = np.concatenate((out, new_vals), axis=0)
+
+    # Create a time series with the target data information
+    # TODO this needs to be some sort of check that allows you to make sure that there is only 1 recipe being stitched
+    # select monthly, daily, or annual data.
+    start = str(min(rp["target_start_yr"]))
+    end = str(max(rp["target_end_yr"]))
+
+    # Note that the pd.date_range call need the date/month defined otherwise it will
+    # truncate the year from start of first year to start of end year which is not
+    # what we want. We want the full final year to be included in the times series.
+    times = pd.date_range(start=start+"-01-01", end=end+"-12-31", freq='M')
+    assert (len(out) == len(times)), "Problem with the length of time"
+
+    # Extract the lat and lon information that will be used to structure the
+    # empty netcdf file.
+    lat = dl[0].lat.values
+    lon = dl[0].lon.values
+
+    # Creat the xarray data set
+    rslt=xr.Dataset({'tas': xr.DataArray(
+        out,
+        coords=[times, lat, lon],
+        dims=["time", "lat", 'lon'],
+        # TODO what other meta data should be added? A path to the recepie file?
+        attrs={'units': 'K',
+               'stitching_id': str(rp['stitching_id'].unique())})
+    })
+
+    return rslt
+
+#############################################################################################
+
+# Read in the recipes!
+recipe = pd.read_csv('./stitches_dev/recpies_for_python.csv')
+
+# Save a copy of the unique file list to minimize the number of files we have to
+# download from pangeo.
+file_list = recipe.file.unique().copy()
+
+# Download all of the files we need for the different recepies. This
+# may take a bit of time to run and WILL NOT WORK IF CONNECTED TO VPN.
+# TODO add some sort of progress bar it is really hard to tell if it is running and it can a couple of mins.
+data_list = get_data_from_pangeo(file_list)
+
+rslts = []
+for id in recipe['stitching_id'].unique():
+    x = recipe[recipe['stitching_id'] == id].copy()
+    out = stitch_gridded(rp=x, dl=data_list, fl=file_list)
+    out.to_netcdf("stitched_"+id+".nc")
+    rslts.append(out)
+
+######################################################################################
```

Comment:
Below is all stuff related to making comparison data and then comparing the results. I doubt that it will go into the python package.